torch/fx/experimental/symbolic_shapes.py

```diff
@@ -348,6 +336,22 @@ def constrain_range(a, *, min: Optional[int], max: Optional[int] = None):
         builtins.max(r.lower, min), builtins.min(r.upper, max)
     )
 
+def constrain_range_eager(a, *, min, max):
```

Comment:
nit: Could you add a comment mentioning that this code path is run in eager so we can't just error whenever we see a concrete integer?

Reply:
Basically what we discussed yesterday. It seemed weird that we were modifying shape_env for concrete integers, but the real reason turned out that we want to be able to execute this op in eager mode, hence we can't just error whenever we see concrete int.

Reply:
Wait, this path is also run when tracing, not just eager. If users constrain a value that is traced to be a static integer this path will also be taken.

Reply:
I feel like naming is basically the hardest part of programming in general :p. Do we want to make a decision between `constrain_on_int ` and `check_range_and_create_symbol `? cc @angelayi , @avikchaudhuri , @tugsbayasgalan, thoughts?

I kind of leaning towards `constrain_on_int` and will add comments in the function to describe more.

Reply:
Fine by me!